docs/markdown/Dependencies.md

```diff
@@ -602,6 +602,41 @@ llvm_dep = dependency('llvm', version : ['>= 8', '< 9'])
 llvm_link = find_program(llvm_dep.get_variable(configtool: 'bindir') / 'llvm-link')
 ```
 
+## Clang
+
+*(since 1.6.0)*
+
+Meson has native support Clang, as well as support for using CMake to find Clang.
+Because of the tight coupling between Clang and LLVM, the Clang dependency has
+specific argument to select the LLVM to use, or an internal version will be used
+(When using the system based finder). This argument is unused with the CMake finder:
+
+```meson
+llvm = dependency('llvm', version : ['>= 16', '< 17'])
+clang = dependency('clang', version : ['>= 16', '< 17'], llvm : llvm, method : 'system')
+```
+
+Both libclang (the C interface) and the C++ interfaces are supported via the
+`language` keyword. The default is to search for the `C` interface.
+
+If the `language` is `c`, then `libclang` will be searched for. This may be
+built static or shared, and is a Clang configuration option.
```

Comment:
Could it be smarter? If the project only has C, search for C. If the project only has C++, search for C++. Else use C.

But maybe that's too much magic for Meson

Reply:
I think that's too much magic for one, and for two the C API seems to be used much more rarely (and to be much less powerful) than the C++ API